### PR TITLE
Adding clang-format, updating Makefile, minor changes to BVH traversal

### PR DESCRIPTION
**Pull request by brandonpelfrey**
Attn @tay10r 

* Created .clang-format to standardize a majority of the formatting.
* Added blank lines after the final line of each member data declaration/method. Personal preference, this leads to better readability as you can jump from block to block of text. In ObjFile.cpp, I've tried to make more logical groupings in terms of line breaks, along the same vein.
* Amended Makefile to also build the ObjFile example
* Changed resolution for CubeOfSpheres to 2048^2
* The `occlusion` flag which was a runtime arg is now one of a set of template flags (more in the future potentially). 
* Removed `w` from Vector3 and changed to `alignas` to get the same effect while avoiding the need to actually compute math on w components.

Let me know your thoughts before I merge these.

### CODE REVIEW

**Comment by ghost** [comment]
Oh snap! Thanks for the heads up. I might need to patch up the other PR I made to include this. I love the changes though! I'll start doing a more in depth review right now.

File: include/FastBVH/BVH.h [hunk 1]
```diff
@@ -20,10 +20,15 @@ template <typename Float>
...
   BBox<Float> bbox;
+
   //! The index of the first primitive.
   uint32_t start;
+
   //! The number of primitives in this node.
   uint32_t nPrims;
+
+  //! Number of elements to skip in flattened tree to get to a left child's sibling. 
+  //! (Node+1 == Node's left child , Node + rightOffset == Node's right child)
```
Comment (ghost) [inline include/FastBVH/BVH.h:31]: This is extremely helpful. I had misunderstood this before and now it's very clear. Thanks!

**Comment by ghost** [comment]
Love the changes! This is a big win for readability. I also like what you did with the `Vector3` and `Traverser` templates. Nothing here that I would want changed.

**Comment by brandonpelfrey** [comment]
> Oh snap! Thanks for the heads up. I might need to patch up the other PR I made to include this. I love the changes though! I'll start doing a more in depth review right now.

I should have probably chatted with you to let you know I'm messing with some of the same code as you right now :smile: I think rebasing my changes to your repo will potentially be easier than vice versa. Do you have a preference?

**Comment by ghost** [comment]
No preference. There might be merge conflicts though.

I figured you might be working on the code as well, so I did this on a `develop` branch in case you wanted to merge in a different branch. Not sure if it's as helpful as I thought it would be though :/

**Comment by brandonpelfrey** [comment]
Merged, let me know if merge conflicts are a pain and we can figure something out.

**Pull request merged**